keps/976-plain-pods/README.md

```diff
@@ -0,0 +1,675 @@
+  annotations:
+    # If the template is left empty, it means that it will match the spec of this pod.
+    kueue.x-k8s.io/pod-group-sets: |-
+      [
+        {
+          name: driver,
+          count: 1,
+        },
+        {
+          name: workers,
+          count: 10,
+          template:
+            spec:
+              containers:
+                - name: worker
+                  requests:
+                    cpu: 1m
+        }
+      ]
+spec:
+  containers:
+  - name: job
+    image: hello-world
+    resources:
+      requests:
+        cpu: 1m
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: job-worker-1
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/pod-group-name: pod-group
+    kueue.x-k8s.io/pod-group-role: worker
+spec:
+  containers:
+  - name: job
+    image: hello-world
+    resources:
+      requests:
+        cpu: 1m
+```
+
+### Notes/Constraints/Caveats (Optional)
+
+#### Skipping Pods belonging to queued objects
+
+Pods owned by jobs managed by Kueue should not be subject to extra management.
+These Pods can be identified based on the ownerReference. For these pods:
+- The webhook should not add a scheduling gate.
+- The pod reconciler should not create a corresponding Workload object.
+
+Note that sometimes the Pods might not be directly owned by a known job object. Here are some
+special cases:
+- MPIJob: The launcher Pod is created through a batch/Job, which is also an known to Kueue, so
+  it's not an issue.
+- JobSet: Also creates Jobs, so not problematic.
+- RayJob: Pods are owned by a RayCluster object, which we don't currently support. This could be
+  hardcoded a known parent, or we could use label selectors for:
+  ```yaml
+  app.kubernetes.io/created-by: kuberay-operator
+  app.kubernetes.io/name: kuberay
+  ```
+
+#### Pods replaced on failure
+
+It is possible that users of plain Pods have a controller for them to handle failures and
+re-creations. These Pods should be able to use the quota that was already assigned to the Workload.
+
+Because Kueue can't know if Pods will be recreated or not, it will hold the entirety of the
+quota until it can determine that the whole Workload finished (all pods are terminated).
+In other words, Kueue won't support [dynamically reclaiming quota](https://github.com/kubernetes-sigs/kueue/issues/78)
+for plain Pods.
+
+### Risks and Mitigations
+
+#### Increased memory usage
+
+In order to support plain Pods, we need to start watching all Pods, even if they are not supposed
+to be managed by Kueue. This will increase the memory usage of Kueue just to maintain the
+informers.
+
+We can use the following mitigations:
+
+1. Drop the unused managedFields field from the Pod spec, like kube-scheduler is doing
+   https://github.com/kubernetes/kubernetes/pull/119556
+2. Apply a selector in the informer to only keep the Pods that have the `kueue.x-k8s.io/managed: true`.
+   The webhook should still affect all Pods.
```

Comment:
Should we restrict Namespace and Pods that perform the webhook processes? In my experience, performing webhook processes against anything would be degrading the K8s cluster's performance. Actually, I often face the issue in the large cluster with Gatekeeper.

Reply:
Let's put namespaceSelector by default to exclude kueue-system and kube-system. Maybe we can make it configurable through helm, although I'm not sure if it's possible.

Reply:
It makes sense. Thanks for the updates.